File: src/brick_game/snake/HighScoreStorage.cpp
//
//  HighScoreStorage.cpp
//  BrickGame
//
//  Created by Alena Ivanova on 10.02.2026.
//

#include "HighScoreStorage.h"
#include <fstream>

void HighScoreStorage::save(int value) const {
    std::ofstream out(fileName_, std::ios::trunc);
    if (out) {
      out << value;
    }
}

int HighScoreStorage::load() const {
    std::ifstream in(fileName_);
    int value = 0;
    if (in) {
      in >> value;
    }
    return value;
}

File: src/brick_game/race/Sources/Engine/HighScoreStorage.swift
//
//  HighScoreStorage.swift
//  Race
//
//  Created by Alena Ivanova on 25.02.2026.
//

import Foundation

struct HighScoreStorage {
    static private let filename: String = "highScoreRace.txt"
    
    static func save(highScore: Int) {
        let fileURL = getFileURL(fileName: filename)
        do {
            try String(highScore).write(to: fileURL, atomically: true, encoding: .utf8)
        } catch {
            print("Failed to save high score: \(error)")
        }
    }
    
    static func get() -> Int {
        let fileURL = getFileURL(fileName: filename)
        do {
            let content = try String(contentsOf: fileURL, encoding: .utf8)
            return Int(content.trimmingCharacters(in: .whitespacesAndNewlines)) ?? 0
        } catch {
            return 0
        }
    }
    
    static private func getFileURL(fileName: String) -> URL {
        let documents = FileManager.default.urls(for: .documentDirectory,
                                                 in: .userDomainMask).first!
        return documents.appendingPathComponent(fileName)
    }
}

